src/Screens/NativeBridgeTests.js

```diff
@@ -52,55 +58,122 @@ export default class nativeBridgeTests extends React.Component {
     }
   };
 
+  getAddress = async () => {
+    try {
+      const mnemonic = await NativeWallet.retrieveMnemonic();
+      let wallet = ethers.Wallet.fromMnemonic(mnemonic);
+      console.log('Address: ', wallet.address);
+      this.setState({address: wallet.address});
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  getBalance = async () => {
+    try {
+      const mnemonic = await NativeWallet.retrieveMnemonic();
+      let wallet = ethers.Wallet.fromMnemonic(mnemonic);
```

Comment:
Re-generating hte wallet form the mnenomic and setting the provider is not really the pattern we want to follow each time we get the balance or send a transation, I think (unless is is VERY cheap)

Can we keep the wallet in some global state, ready to use?

Reply:
Has this been discussed between you and Hao?